File: leetcode/python/118_pascal_triangle.py
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        pascal: list[list[int]] = [[1]]
        for i in range(1, numRows):
            pascal.append([1])
            for j in range(1, i):
                pascal[-1].append(pascal[i-1][j-1]+pascal[i-1][j])
            pascal[-1].append(1)

        return pascal

s = Solution()
print(s.generate(5))

File: leetcode/rust/src/118_pascal_triangle.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut res = vec![vec![1]];
        for i in 1..(num_rows as usize) {
            let mut row = vec![1];
            for j in 1..i {
                row.push(res[i - 1][j - 1] + res[i - 1][j]);
            }
            row.push(1);
            res.push(row);
        }
        res
    }
}